install.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at:
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the
+# License.
+
+# Python 2/3 compatibility
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import platform
+import shutil
+import sys
+from subprocess import check_call
+from os.path import join, abspath, isdir, dirname
+
+_PYPY = hasattr(sys, 'pypy_translation_info')
+_OS = platform.system()
+_WIN = _OS == 'Windows'
+_MAC = _OS == 'Darwin'
+_LINUX = _OS == 'Linux'
+
+_C_EXT_DEPENDENCY_DIR = abspath(join(dirname(os.path.abspath(__file__)), 'ion-c-build'))
+_C_EXT_DEPENDENCY_LIB_LOCATION = abspath(join(_C_EXT_DEPENDENCY_DIR, 'lib'))
+_C_EXT_DEPENDENCY_INCLUDES_LOCATIONS = abspath(join(_C_EXT_DEPENDENCY_DIR, 'include'))
+
+_IONC_REPO_URL = "https://github.com/amzn/ion-c.git"
+_IONC_DIR = abspath(join(dirname(os.path.abspath(__file__)), 'ion-c'))
+_IONC_LOCATION = abspath(join(dirname(os.path.abspath(__file__)), 'ion-c', 'build', 'release'))
+_IONC_INCLUDES_LOCATIONS = {
+    'ionc': abspath(join(dirname(os.path.abspath(__file__)), 'ion-c', 'ionc', 'include', 'ionc')),
+    'decNumber': abspath(join(dirname(os.path.abspath(__file__)), 'ion-c', 'decNumber', 'include', 'decNumber'))
+}
+
+_LIB_PREFIX = 'lib'
+
+_LIB_SUFFIX_MAC = '.dylib'
+_LIB_SUFFIX_WIN = '.lib'
+_LIB_SUFFIX_LINUX = '.so'
+
+
+def _get_lib_name(name):
+    if _MAC:
+        return '%s%s%s' % (_LIB_PREFIX, name, _LIB_SUFFIX_MAC)
+    elif _LINUX:
+        return '%s%s%s' % (_LIB_PREFIX, name, _LIB_SUFFIX_LINUX)
+    elif _WIN:
+        return '%s%s' % (name, _LIB_SUFFIX_WIN)
+
+
+def _library_exists():
+    return _library_exists_helper('ionc') and _library_exists_helper('decNumber')
+
+
+def _library_exists_helper(name):
+    return os.path.exists(join(_C_EXT_DEPENDENCY_LIB_LOCATION, _get_lib_name(name)))
+
+
+def _download_ionc():
```

Comment:
As discussed offline, let's use an ion-c module so it can be pinned to a particular commit. In this script we'd need to make sure it's checked out, then use the same build logic.

Reply:
Will work on this in next step.